File: go/problems/day15_test.go
package problems

import "testing"

var day15 Solvable[int] = Day15{}

func TestDay15_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day15_1")
	result := day15.Solve(input)

	var expected int = 1320
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay15_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day15_2")
	result := day15.Solve(input)

	var expected int = 505379
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day15.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day15.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 15", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day15_1");
    const result = solve(input);
    assert.equal(result, 1320);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day15_2");
    const result = solve(input);
    assert.equal(result, 505379);
  });
});
